### DISCUSSION TITLE
Placing PHP Code on Pages Fails

### DISCUSSION THREAD

**Discussion by im-L0r3** [e107inc/e107#4763] [General]
The goal is to create pages where I can show users snippets of PHP code.

I can put some code on the page, but some bits of specific code either "disappear" when pasting, or the page contents is deleted when hitting the save button.

I can imagine this is some sort of conflict with e107s PHP, but is there any sort of workaround? Iv attempting escaping in various ways without success.

THIS WORKS:
`$curl = curl_init();
curl_setopt_array($curl, array(
CURLOPT_URL => https://api.nightbot.tv/oauth2/token,
CURLOPT_SSL_VERIFYPEER => false,
CURLOPT_RETURNTRANSFER => true,
CURLOPT_POST => true,
CURLOPT_POSTFIELDS => "grant_type=client_credentials&client_id={YourClientIdHere}&client_secret={YourClientSecretHere}&scope=channel channel_send commands commands_default regulars subscribers timers"
));
$nightbotApi = curl_exec`

BUT THE FULL CODE FAILS to paste at the spot immediately after the "url_exec" above:
`$curl = curl_init();
curl_setopt_array($curl, array(
CURLOPT_URL => https://api.nightbot.tv/oauth2/token,
CURLOPT_SSL_VERIFYPEER => false,
CURLOPT_RETURNTRANSFER => true,
CURLOPT_POST => true,
CURLOPT_POSTFIELDS => "grant_type=client_credentials&client_id={YourClientIdHere}&client_secret={YourClientSecretHere}&scope=channel channel_send commands commands_default regulars subscribers timers"
));
$nightbotApi = curl_exec($curl);
curl_close($curl);`

**Comment by CaMer0n** [comment]
@im-L0r3 Are you wrapping the code in [code] or [pre] BB codes or using similar in TinyMce?

**Reply by im-L0r3** [reply]
Thank for your response! Using [code] or [pre] tags either via inserting them via the UI or by editing the source still causes the pasted php code to "disappear". The UI just wont allow it to paste, and inserting it manually in the source just blanks the page back to default when saving.

Iv seen TinyMce in out of the corner of my eye somewhere in e107 but Im not familiar with it. Ill go check it out.

**Reply by im-L0r3** [reply]
In TinyMce, I toggled "Paste as text by default" and tried with it both off and on with the same result.

**Reply by im-L0r3** [reply]
I can rewrite the code in BASH and it will paste fine, as Im just giving the readers general concepts of how to do "stuff".
It just bugs my inner nerd to use a work around rather than solving this.

**Reply by CaMer0n** [reply]
I'll see what can be done to fix that. I believe whatever caused this was implemented as a security measure in the past.
